### PR TITLE
Chore/remove for loops

### PR DESCRIPTION
**Pull request by DerekMaffett**
Removes `for` loops that could be substituted for simple `map/reduce` functions. A few still remain due to being complicated algorithms. If appropriate, we can refactor those later, but they're not as straightforward.

### CODE REVIEW

File: src/components/victory-bar/victory-bar.js [hunk 1]
```diff
@@ -141,17 +141,19 @@ class VictoryBar extends React.Component {
...
-    const labelComponents = [];
-    for (let index = 0, len = this.dataKeys.length; index < len; index++) {
+
+    const dataComponents = this.dataKeys.map((_dataKey, index) => {
       const dataProps = this.getComponentProps(dataComponent, "data", index);
-      dataComponents[index] = React.cloneElement(dataComponent, dataProps);
+      return React.cloneElement(dataComponent, dataProps);
+    });
 
+    const labelComponents = compact(this.dataKeys.map((_dataKey, index) => {
```
Comment (ryan-roemer) [inline src/components/victory-bar/victory-bar.js]: GLOBAL NOTE: If we don't need to take a dependency on `compact`, then we can simply do this natively with:

```js
const dataComponents = this.dataKeys
  .map((_dataKey, index) => { /* RETURN EMPTY OR VALUE */ })
  .filter(Boolean);
```

Comment (boygirl) [inline src/components/victory-bar/victory-bar.js]: @DerekMaffett I believe we're using `compact` in either data or domain helpers, but I'm happy to move to this native pattern and drop the dependency.

Comment (DerekMaffett) [inline src/components/victory-bar/victory-bar.js]: I used it for the sake of readability, but yes, `compact` is technically just an identity filter. I can change it to that.

**Comment by boygirl** [comment]
@DerekMaffett approved pending `compact` change

**Pull request merged**